ArcGISRuntimeSDKQt_CppSamples/Analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.h

```diff
@@ -0,0 +1,115 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifndef CONFIGURESUBNETWORKTRACE_H
+#define CONFIGURESUBNETWORKTRACE_H
+
+namespace Esri
+{
+namespace ArcGISRuntime
+{
+class Error;
+class UtilityTraceCondition;
+class UtilityElement;
+class UtilityNetwork;
+class UtilityNetworkDefinition;
+class UtilityTerminal;
+class UtilityTier;
+class UtilityTraceParameters;
+class UtilityTraceConditionalExpression;
+class UtilityTraceConfiguration;
+}
+}
+
+#include <QObject>
+#include <QUrl>
+#include <QUuid>
+
+#include "UtilityNetworkAttributeComparison.h"
+#include "UtilityNetworkAttribute.h"
+
+class ConfigureSubnetworkTrace : public QObject
+{
+  Q_OBJECT
+
+  Q_PROPERTY(bool busy MEMBER m_busy NOTIFY busyChanged)
+  Q_PROPERTY(bool dialogVisible MEMBER m_dialogVisible NOTIFY dialogVisibleChanged)
+  Q_PROPERTY(bool textFieldVisible MEMBER m_textFieldVisible NOTIFY textFieldVisibleChanged)
+  Q_PROPERTY(QString dialogText MEMBER m_dialogText NOTIFY dialogTextChanged)
+  Q_PROPERTY(QString expressionBuilder MEMBER m_expressionBuilder NOTIFY expressionBuilderChanged)
+  Q_PROPERTY(QStringList attributeListModel MEMBER m_attributeListModel NOTIFY attributeListModelChanged)
+  Q_PROPERTY(QStringList conditionBarrierExpressionListModel MEMBER m_conditionBarrierExpressionListModel NOTIFY conditionBarrierExpressionChanged)
+  Q_PROPERTY(QStringList valueSelectionListModel MEMBER m_valueSelectionListModel NOTIFY valueSelectionListModelChanged)
+
+public:
+  explicit ConfigureSubnetworkTrace(QObject* parent = nullptr);
+  ~ConfigureSubnetworkTrace();
+
+  static void init();
+
+  Q_INVOKABLE void addCondition(const QString& selectedAttribute, int selectedOperator, const QVariant& selectedValue);
+  Q_INVOKABLE void changeIncludeBarriersState(bool includeBarriers);
+  Q_INVOKABLE void changeIncludeContainersState(bool includeContainers);
+  Q_INVOKABLE void codedValueOrInputText(const QString& currentText);
+  Q_INVOKABLE void reset();
+  Q_INVOKABLE void trace();
+
+signals:
+  void attributeListModelChanged();
+  void busyChanged();
+  void conditionBarrierExpressionChanged();
+  void dialogTextChanged();
+  void dialogVisibleChanged();
+  void expressionBuilderChanged();
+  void textFieldVisibleChanged();
+  void valueSelectionListModelChanged();
+
+private slots:
+  void onTraceCompleted();
+  void onUtilityNetworkLoaded(const Esri::ArcGISRuntime::Error& e);
+
+private:
+
+  QString comparisonOperatorToString(const Esri::ArcGISRuntime::UtilityAttributeComparisonOperator& comparisonOperator) const;
+  QString expressionToString(Esri::ArcGISRuntime::UtilityTraceConditionalExpression* expression) const;
+  QVariant convertToDataType(const QVariant& value, const Esri::ArcGISRuntime::UtilityNetworkAttributeDataType& dataType);
```

Comment:
Optional, but I don't think any of these actually need to be member functions. They are all stateless, so could easily be static helpers.

Possibly not `convertToDataType` since it is not const, but unsure about that.